### DISCUSSION TITLE
Rich markup in Zsh completion help lines

### DISCUSSION THREAD

**Discussion by tiangolo** [fastapi/typer#821] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorials in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer
from typing_extensions import Annotated

app = typer.Typer(rich_markup_mode="rich")


@app.command()
def hello(
    name: Annotated[
        str, typer.Option(help="The [bold]cool[/bold] name of the [green]user[/green]")
    ],
):
    """
    Say [bold]hi[/bold] to the user.
    """
    print(f"Hello {name}")


@app.command()
def goodbye(
    name: Annotated[
        str, typer.Option(help="The [bold]cool[/bold] name of the [green]user[/green]")
    ],
):
    """
    Say [bold]goodbye[/bold] to the user.
    """
    print(f"Goodbye {name}")

if __name__ == "__main__":
    app()
```

### Description

* Create this script in `main.py`
* In Zsh, start running it with the `typer` command, start typing: `typer main.py run `, then hit <kbd>tab</kbd> to trigger autocompletion, it shows the two commands, which is correct:

* `goodbye`
* `hello`

But their Zsh help string includes the Rich markup content:

```
$ typer main.py run <tab>

goodbye  -- Say [bold]goodbye[/bold] to the user.
hello    -- Say [bold]hi[/bold] to the user.
```

![Screenshot from 2024-05-02 15-12-44](https://github.com/tiangolo/typer/assets/1326112/e0f6ad09-6a2a-496f-b1c5-7184f302b8b1)

It should show plaintext without `[bold]` markup parts... or better, it should render those parts and show them in actual green color (using the terminal color codes, etc, the stuff Rich does).

### Operating System

Linux, macOS

### Operating System Details

This applies mainly to Zsh, the behavior might be similar or different in PowerShell. It only applies in shells that can show autocompletion of commands with help sections.

### Typer Version

Typer version: 0.12.3

### Python Version

any

**Comment by svlandeg** [comment, marked as answer]
I submitted a PR https://github.com/tiangolo/typer/pull/877 that at least should adress removing the tags from the output.

If anyone wants to test the PR, I'd appreciate it! Just let me know which OS and shell you're on, and fill in the following boxes for your system:

- [ ] Shows Rich tags in completion text with `master`
- [ ] Removes Rich tags in completion text with PR 877
- [ ] Shell is able to show Rich formatting in simple script
- [ ] Shows Rich formatting in completion text with PR 877

A simple script you can use to double check whether your console supports Rich formatting, is the following:
```
    from rich.console import Console

    description = "The [bold]name[/bold] of the [green]user[/green]"

    console = Console()
    console.print(description)
```
